server/routerlicious/packages/lambdas/src/scribe/lambda.ts

```diff
@@ -612,17 +612,22 @@ export class ScribeLambda implements IPartitionLambda {
 			return;
 		}
 		let databaseCheckpointFailed = false;
-		this.pendingP = clearCache
-			? this.checkpointManager.delete(this.protocolHead, true)
-			: this.writeCheckpoint(checkpoint).catch((error) => {
-					databaseCheckpointFailed = true;
-					Lumberjack.error(
-						`Error writing database checkpoint.`,
-						getLumberBaseProperties(this.documentId, this.tenantId),
-						error,
-					);
-			  });
-		this.pendingP
+
+		this.pendingP = (
+			clearCache
+				? this.checkpointManager.delete(this.protocolHead, true)
+				: this.writeCheckpoint(checkpoint)
+		)
+			.catch((error) => {
```

Comment:
This will have an interesting behavior where after the delete/write error is caught, we will still do the `then`. The `then` could then hypothetically error on one of the `.checkpoint()` or `.checkpointCore()` calls, which would be a different error.

We need to be careful here. Existing behavior would only perform `checkpoint` after _write_ failure; but this new logic will also perform `checkpoint` after _delete_ failure.

Reply:
Thank you for finding this. I corrected the behavior